### PR TITLE
Before epoch fix

### PR DESCRIPTION
**Pull request by kupci**

### CODE REVIEW

File: datetime/src/test/java/com/fasterxml/jackson/datatype/jsr310/TestZonedDateTimeSerialization.java [hunk 1]
```diff
@@ -751,6 +763,15 @@ public void testNumericCustomPatternWithAnnotations() throws Exception
         assertEquals(input.value.toInstant(), result.value.toInstant());
     }
 
+    @Test
+    public void testInstantPriorToEpochIsEqual() throws Exception
+    {
+        final Instant original = Instant.ofEpochMilli(-1);
+        final String serialized = MAPPER.writeValueAsString(original);
+        final Instant deserialized = MAPPER.readValue(serialized, Instant.class);
+        assertEquals(true,original.compareTo(deserialized)==0);
```
Comment (pjfanning) [inline datetime/src/test/java/com/fasterxml/jackson/datatype/jsr310/TestZonedDateTimeSerialization.java]: could this be just `assertEquals(original, deserialized)`?

**Comment by pjfanning** [comment]
@kupci could you add the test case in https://github.com/FasterXML/jackson-modules-java8/issues/120#issuecomment-520986313 (if this does fox #120)?

**Comment by kupci** [comment]
> @kupci could you add the test case in [#120 (comment)](https://github.com/FasterXML/jackson-modules-java8/issues/120#issuecomment-520986313) (if this does fix #120)?

Yes, I've got that test case from the issue included in the test suite.

**Comment by pjfanning** [comment]
@cowtowncoder this PR looks good to me - do you think it is ok to merge it?

**Comment by cowtowncoder** [comment]
@pjfanning Ok. So, to get this 2.10 I can... I think I'll merge this, just backport.

**Comment by cowtowncoder** [comment]
Finally got this merged: thank you everyone for getting this annoying long-time bug fixed.

**Pull request merged**